### PR TITLE
Allow the constant initialization pass to speculatively execute Traspose operations.

### PR DESCRIPTION
**Pull request by zdevito**
This allows some operations to be lifted out of control flow and into the init net.

### CODE REVIEW

File: onnx/optimizer/passes/split.h [hunk 1]
```diff
@@ -11,9 +11,68 @@ static constexpr const char* impure_operators[] = {
...
+  for(Value* input : graph.inputs()) {
+    capture_map[input->uniqueName()] = input;
+  }
+  for(auto it = graph.nodes().begin(), end = graph.nodes().end();
+      it != end;
+      ++it) {
+    Node * n = *it;
+
+    // prevent overwriting entries in the capture map with the using capture nodes
+    if(n->kind() == Symbol("Captured")) {
```
Comment (jamesr66a) [inline onnx/optimizer/passes/split.h]: Can you use kCaptured here?

File: onnx/optimizer/passes/split.h [hunk 2]
```diff
@@ -11,9 +11,68 @@ static constexpr const char* impure_operators[] = {
...
+      continue;
+    }
+
+    for(Value* o : n->outputs() ) {
+      capture_map[o->uniqueName()] = o;
+    }
+
+    for(auto s : n->attributeNames()) {
```
Comment (jamesr66a) [inline onnx/optimizer/passes/split.h]: Can you just use `DescendOnGraphAttributes` here?

https://github.com/onnx/onnx/blob/master/onnx/optimizer/passes/optimize_pass.h#L30

You can capture the capture map in a lambda

File: onnx/test/optimizer_test.py [hunk 3]
```diff
@@ -425,6 +425,89 @@ def test_split(self):
...
+            [node] + if_stmt,
+            'test-optimize-split',
+            [],
+            [helper.make_tensor_value_info('output', TensorProto.FLOAT, (1,))])
+
+        def opt(graph, opts):
+            orig_model = helper.make_model(graph, producer_name='onnx-test')
+            return onnx.optimizer.optimize(orig_model, opts).graph
+
+        def pretty(g):
```
Comment (jamesr66a) [inline onnx/test/optimizer_test.py]: lol

**Comment by zdevito** [comment]
Closing in favor of doing this transform before export from PyTorch.

**Pull request closed without merging**